src/gui/gt_qmlaction.h

```diff
@@ -23,35 +25,44 @@ class GT_GUI_EXPORT GtQmlAction : public QObject
 {
     Q_OBJECT
 
-    Q_PROPERTY(QString text READ text WRITE setText NOTIFY textChanged)
+    Q_PROPERTY(QString text READ text NOTIFY textChanged)
     Q_PROPERTY(QUrl iconUrl READ iconUrl WRITE setIconUrl NOTIFY iconUrlChanged)
     Q_PROPERTY(bool enabled READ enabled WRITE setEnabled NOTIFY enabledChanged)
+    Q_PROPERTY(QString tooltip READ tooltip NOTIFY tooltipChanged)
+    Q_PROPERTY(bool visible READ visible WRITE setVisible NOTIFY visibleChanged)
 public:
     explicit GtQmlAction(QObject* parent = nullptr);
-    explicit GtQmlAction(QString text, QObject* parent = nullptr);
-    GtQmlAction(QString text, QUrl iconUrl, QObject* parent = nullptr);
+    GtQmlAction(QString text, QUrl icon, QObject* parent = nullptr);
+
+    ~GtQmlAction() override;
 
     QString text() const;
     void setText(const QString& text);
 
     QUrl iconUrl() const;
-    void setIconUrl(const QUrl&url);
+    void setIconUrl(const QUrl& url);
+
+
+    QString tooltip() const;
+    void setTooltip(QString t);
```

Comment:
Just as a suggestion (as I have noticed this recently): Qt uses `toolTip` throughout its API. I propose we should stick to the same naming convetion (even though I would have named the function all lower case ass well intuitively). I'll mark only these functions here, of couse this applies to all symbols named `tooltip`.
```suggestion
    QString toolTip() const;
    void setToolTip(QString t);
```

Reply:
Good suggestion. I will change it.